etc.c

```diff
@@ -389,6 +389,7 @@ checkType(Str s, Lineprop **oprop, Linecolor **ocolor)
 			else {
 			    Strshrink(s, plen);
 			    prop -= plen;
+			    prop = prop < prop_buffer ? prop_buffer : prop;
```

Comment:
Strictly speaking, a pointer going out of bound (with one exception of 1 past the end) is already UB [^0]. Instead of trying to check for UB and then correcting it, it's better to check `plen` before subtracting.

```c
if (prop - prop_buffer < plen) // would underflow
    prop = prop_buffer;
else
    prop -= plen;
```

[^0]: https://port70.net/~nsz/c/c99/n1256.html#6.5.6p8